pandas/core/frame.py

```diff
@@ -1141,7 +1142,15 @@ def __len__(self) -> int:
         """
         return len(self.index)
 
-    def dot(self, other):
+    @overload
```

Comment:
why is this ignore?

Reply:
i see your comment, can you make this the actual type error as a comment

Reply:
Done